extension/source/components/Button.tsx

```diff
@@ -20,6 +21,10 @@ const Button = (props: {
     classes.push('loading');
   }
 
+  if (props.disabled) {
+    classes.push('disabled');
+  }
+
   return (
     <div
       className={classes.join(' ')}
```

Comment:
```suggestion
      className={`button ${highlight && "highlight"} ${loading && "loading"} ${disabled && "disabled"}`}
```

Will keep this component stateless

Reply:
Similar to before, I don't like this because it inserts `false`.

I think this pattern comes up a lot though, and I agree that the `classes.join` approach is a bit verbose. How about a utility like this?

```ts
export default function classFlags(classes: Record<string, boolean>) {
  return Object.entries(classes)
    .filter(([, flag]) => flag)
    .map(([className]) => className)
    .join(' ');
}
```

That way we could write:

```tsx
<div className={classFlags({ button: true, highlight, loading, disabled })}/>
```

Reply:
> In other components, conditional styles must set via state mutation and not local variable mutation.

I'm not sure why but I'm not able to follow what you're getting at here. Maybe an example where something similar to what I'm doing goes badly would help?

Reply:
apologies for not being clear, I meant that
state variables = variables declared using useState method
local variables = any other variable declared in the body of the component

Also I played around in a fresh project and I think that the reason I gave is wrong. Instead of re-rendering every child component, it wont do anything if a local variable is mutated. 

Btw this doesn't apply in this particular case anymore, so I guess let's come back to this later if it comes back.

Reply:
👍